src/plugins/plugin.legend.js

```diff
@@ -90,8 +90,8 @@ defaults._set('global', {
  * @return {Number} width of the color box area
  */
 function getBoxWidth(labelOpts, fontSize) {
-	return labelOpts.usePointStyle ?
-		labelOpts.pointSize || fontSize * Math.SQRT2 :
+	return labelOpts.usePointStyle && labelOpts.boxWidth > fontSize ?
+		fontSize * Math.SQRT2 :
```

Comment:
I think `Math.SQRT2` is enough in `draw` and should be removed from here (probably the cause for originally too big points)

Reply:
Tested both with and without, and works fine. I will remove and commit.